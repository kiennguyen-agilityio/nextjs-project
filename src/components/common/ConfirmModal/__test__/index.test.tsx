import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import ConfirmationModal from '..';

describe('ConfirmationModal component', () => {
  test('does not render when isOpen is false', () => {
    render(
      <ConfirmationModal
        isOpen={false}
        onConfirm={() => {}}
        onCancel={() => {}}
        message="Are you sure?"
      />,
    );
    expect(screen.queryByText('Are you sure?')).not.toBeInTheDocument();
  });

  test('renders when isOpen is true', () => {
    render(
      <ConfirmationModal
        isOpen={true}
        onConfirm={() => {}}
        onCancel={() => {}}
        message="Are you sure?"
      />,
    );
    expect(screen.getByText('Are you sure?')).toBeInTheDocument();
  });

  test('calls onCancel when Cancel button is clicked', () => {
    const handleCancel = jest.fn();
    render(
      <ConfirmationModal
        isOpen={true}
        onConfirm={() => {}}
        onCancel={handleCancel}
        message="Are you sure?"
      />,
    );

    fireEvent.click(screen.getByText('Cancel'));

    waitFor(() => {
      expect(handleCancel).toHaveBeenCalledTimes(1);
    });
  });

  test('calls onConfirm and manages loading state when Confirm button is clicked', async () => {
    const handleConfirm = jest.fn(
      () => new Promise((resolve) => setTimeout(resolve, 1000)),
    );

    render(
      <ConfirmationModal
        isOpen={true}
        onConfirm={handleConfirm}
        onCancel={() => {}}
        message="Are you sure?"
      />,
    );

    fireEvent.click(screen.getByText('Confirm'));

    await waitFor(() => {
      expect(handleConfirm).toHaveBeenCalledTimes(1);
      expect(screen.getByText('Confirm')).not.toHaveAttribute('disabled');
    });
  });

  test('does not render buttons if loading is true', async () => {
    const { container } = render(
      <ConfirmationModal
        isOpen={true}
        onConfirm={async () => {}}
        onCancel={() => {}}
        message="Are you sure?"
      />,
    );

    fireEvent.click(screen.getByText('Confirm'));

    await waitFor(() => {
      expect(container.querySelector('button[disabled]')).toBeInTheDocument();
    });
  });

  test('matches the snapshot', () => {
    const { asFragment } = render(
      <ConfirmationModal
        isOpen={true}
        onConfirm={() => {}}
        onCancel={() => {}}
        message="Are you sure?"
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
