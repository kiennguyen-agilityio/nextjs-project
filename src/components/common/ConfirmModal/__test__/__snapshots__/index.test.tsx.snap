// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationModal component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="fixed inset-0 flex items-center justify-center bg-gray-500 bg-opacity-50"
  >
    <div
      class="bg-white p-6 rounded-lg shadow-lg"
    >
      <p
        class="mb-4"
      >
        Are you sure?
      </p>
      <div
        class="flex justify-end"
      >
        <button
          aria-label="button"
          class="flex items-center rounded p-2 bg-white text-[#62656e] hover:bg-gray-300 mr-2"
          type="button"
        >
          Cancel
        </button>
        <button
          aria-label="button"
          class="flex items-center rounded p-2 bg-rose-500 text-white hover:bg-rose-600 "
          type="button"
        >
          Confirm
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
